src/components/textBlock/TextBlock.js

```diff
@@ -6,6 +6,17 @@ export class TextBlock extends Component {
     hidden: true
   };
 
+  componentDidMount() {
+    const { initialText } = this.props;
+
+    console.log(initialText);
+  }
+
+  componentDidUpdate(prevProp, prevState) {
+    const { initialText } = this.props;
+    if (!prevProp.initialText && initialText) this.setState({inputText: initialText});
```

Comment:
and what if `prevProp.initialText` was _'one text'_ and `this.props.initialText` became _'other text'_? How is state updated if props was changed?
You have to compare props and if it is different, then change state.